pymodbus/client/base.py

```diff
@@ -251,6 +269,10 @@ async def __aenter__(self):
 
     async def __aexit__(self, klass, value, traceback):
         """Implement the client with aexit block."""
+        cancel_pending = False
+        if klass is asyncio.CancelledError:
+            cancel_pending = True
+        await self.drain(cancel_pending)
```

Comment:
The needs to go into close ! Even though an app calls client.close() there are no limitation that the running loop is not continued, and the client object reused.